source/slides/02_linux_basics.rst

```diff
@@ -0,0 +1,508 @@
+.. _02_linux_basics:
+
+Linux Basics (Day 2)
+====================
+
+Topics
+------
+
+* Packages (leftover from Wednesday)
+* Editors
+* Git
+* Openstack Setup
+
+Editors
+=======
+
+Text Editors
+------------
+
+.. note::
+
+    Quick intro/history:  ed editor
+    Pros: low-bandwidth, installed pretty much everywhere, very fast and powerful
+    for complicated and repetitive tasks
+    Cons: Steep learning curve, different “modes” can be confusing at first
+    Sublime and other desktop editors: nice for serious programming, but learn
+    the basics of simple text editors even if you want to be a developer, you
+    won't always be able to edit your code on your own desktop
+
+* ed -> Vi -> Vim
+* Stallman -> lisp -> emacs
+
+.. figure:: ../_static/xkcd_378.png
+    :align: center
+    :scale: 85%
+
+Avoid Pico/Nano, Notepad++, SublimeText
+
+Emacs
+-----
+
+.. figure:: ../_static/emacs_logo.jpeg
+    :align: right
+
+.. note::
+
+    Originally released 1976
+    name from Editor MACros for TECO editor, originally Tape Editor and
+    COrrector at MIT in the '60s
+
+*But, along the way, I wrote a text editor, Emacs. The interesting idea about
+Emacs was that it had a programming language, and the user's editing commands
+would be written in that interpreted programming language, so that you could
+load new commands into your editor while you were editing. You could edit the
+programs you were using and then go on editing with them.*
+
+ -- Richard Stallman, http://www.gnu.org/gnu/rms-lisp.html
+
+Vim
+---
+
+.. figure:: ../_static/vim_logo.jpeg
+    :align: right
+
+.. note::
+
+    originally written for Amiga systems (Commodore PCs), 1988
+    vim released 1991
+    vimscript, Lua (as of Vim 7.3), Perl, Python, Racket, Ruby, Tcl (tool
+    command language).
+    vi written by Bill Joy in 1976, visual mode for line editor called ex
+    line editors are from age of teleprinters, no cursors
+
+* Available almost everywhere
+* Lightweight
+* Design decisions explained in http://docs.freebsd.org/44doc/usd/12.vi/paper.html
+* Modal editor (command, insert, visual)
+
+How to choose
+-------------
+
+* What can the people around you help with?
+* Try both; choose one and get good at it
+* Have a good answer when people ask why you made that choice
+    * "Because it's familiar" is tolerated
```

Comment:
Might be good to use italics for all of these quotes to give it a little contrast.